ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/Rao.java

```diff
@@ -0,0 +1,197 @@
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.powsybl.commons.Versionable;
+import com.powsybl.commons.config.PlatformConfig;
+import com.powsybl.commons.util.ServiceLoaderCache;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.computation.local.LocalComputationManager;
+import com.powsybl.iidm.network.Network;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * RA optimisation main API. It is a utility class (so with only static methods) used as an entry point for running
+ * a Ra optimisation allowing to choose either a specific find implementation or just to rely on default one.
+ *
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
+ */
+public final class Rao {
+
+    private Rao() {
+        throw new AssertionError("Utility class should not been instantiated");
+    }
+
+    private static final Supplier<List<RaoProvider>> PROVIDERS_SUPPLIERS
+            = Suppliers.memoize(() -> new ServiceLoaderCache<>(RaoProvider.class).getServices());
+
+    /**
+     * A RA optimisation runner is responsible for providing convenient methods on top of {@link RaoProvider}:
+     * several variants of synchronous and asynchronous run with default parameters.
+     */
+    public static class Runner implements Versionable {
+
+        private final RaoProvider provider;
+
+        public Runner(RaoProvider provider) {
+            this.provider = Objects.requireNonNull(provider);
+        }
+
+        public CompletableFuture<RaoComputationResult> runAsync(Network network, Crac crac, String workingStateId, ComputationManager computationManager, RaoParameters parameters) {
+            Objects.requireNonNull(workingStateId);
+            Objects.requireNonNull(parameters);
+            return provider.run(network, crac, workingStateId, computationManager, parameters);
+        }
+
+        public CompletableFuture<RaoComputationResult> runAsync(Network network, Crac crac, ComputationManager computationManager, RaoParameters parameters) {
+            return runAsync(network, crac, network.getVariantManager().getWorkingVariantId(), computationManager, parameters);
+        }
+
+        public CompletableFuture<RaoComputationResult> runAsync(Network network, Crac crac, RaoParameters parameters) {
+            return runAsync(network, crac, LocalComputationManager.getDefault(), parameters);
+        }
+
+        public CompletableFuture<RaoComputationResult> runAsync(Network network, Crac crac) {
+            return runAsync(network, crac, RaoParameters.load());
+        }
+
+        public RaoComputationResult run(Network network, Crac crac, String workingStateId, ComputationManager computationManager, RaoParameters parameters) {
+            Objects.requireNonNull(workingStateId);
+            Objects.requireNonNull(parameters);
```

Comment:
Same as before, Objects.requireNonNull on network and Crac

Reply:
done